txn.go

```diff
@@ -241,18 +293,37 @@ func (txn *Txn) Delete(table string, obj interface{}) error {
 		indexTxn := txn.writableIndex(table, name)
 
 		// Handle the update by deleting from the index first
-		ok, val, err := indexSchema.Indexer.FromObject(existing)
-		if err != nil {
-			return fmt.Errorf("failed to build index '%s': %v", name, err)
-		}
-		if ok {
-			// Handle non-unique index by computing a unique index.
-			// This is done by appending the primary key which must
-			// be unique anyways.
-			if !indexSchema.Unique {
-				val = append(val, idVal...)
+		switch indexer := indexSchema.Indexer.(type) {
+		case SingleIndexer:
+			ok, val, err := indexer.FromObject(existing)
+			if err != nil {
+				return fmt.Errorf("failed to build index '%s': %v", name, err)
+			}
+			if ok {
+				// Handle non-unique index by computing a unique index.
+				// This is done by appending the primary key which must
+				// be unique anyways.
+				if !indexSchema.Unique {
+					val = append(val, idVal...)
+				}
+				indexTxn.Delete(val)
+			}
+		case MultiIndexer:
+			ok, vals, err := indexer.FromObject(existing)
```

Comment:
Same here, just life the variables out of the switch to share the logic, since the multi-value case is a superset